### PR TITLE
Codex SDK migration + split agent/model selectors

### PR DESCRIPTION
**Pull request by Connoropolous**
## Summary
- switch `packages/codex-runner` from raw `codex exec --json` parsing to `@openai/codex-sdk` typed streaming events
- map `appendSystemPrompt` to Codex `developer_instructions`
- split runner selection into independent `agent` and `model` selectors
- add issue-description tags for runner/model override: `[agent=...]` and `[model=...]`
- keep session pinning behavior for resumed sessions and add conflict handling (explicit agent wins over conflicting model)
- add tests for label+description selector behavior in `packages/edge-worker/test/EdgeWorker.runner-selection.test.ts`

## Validation
- `pnpm --filter cyrus-edge-worker typecheck`
- `pnpm exec vitest run test/EdgeWorker.runner-selection.test.ts` (from `packages/edge-worker`)
- `pnpm --filter cyrus-codex-runner typecheck`
- `pnpm --filter cyrus-codex-runner build`
- `pnpm --filter cyrus-edge-worker build`
- `pnpm --filter cyrus-f1 build`

## F1 smoke checks
- labels scenario: `orchestrator,codex,gpt-5-codex` -> logs show `Model override via selector: gpt-5-codex` and `runner selection ... codex`; `view-session` shows `Using model: gpt-5-codex`
- description scenario: labels `orchestrator` + description `[agent=codex] [model=gpt-5.2-codex]` -> logs show `Model override via selector: gpt-5.2-codex` and `runner selection ... codex`; `view-session` shows `Using model: gpt-5.2-codex`

---

> [!NOTE]
> **Medium Risk**
> Introduces new runner implementations and changes runner/model selection and config migrations, which can affect session routing and execution behavior across providers; tests and back-compat migrations reduce but don’t eliminate integration risk.
> 
> **Overview**
> Cyrus now supports **Cursor as a first-class runner** and migrates the **Codex runner** from ad-hoc CLI parsing to `@openai/codex-sdk` streamed events, emitting consistent `tool_use`/`tool_result` messages so Linear timelines show command/file/web/MCP/todo activity instead of only final output.
> 
> Runner selection is expanded to support independent `[agent=...]` and `[model=...]` description tags (with clear precedence over labels and model-based runner inference), and session tracking/types are extended to persist `codexSessionId`/`cursorSessionId`.
> 
> Config handling is updated to use runner-specific defaults (`claudeDefaultModel`/`claudeDefaultFallbackModel` plus `geminiDefaultModel` and `codexDefaultModel`), including migrations/back-compat in CLI/config-updater and updated F1 test harness config; repository docs/skills are reorganized to centralize the F1 test-drive protocol via shared skills/symlinks, with new offline and live test-drive reports/scripts validating Cursor behavior.
> 
> <sup>Written by [Cursor Bugbot](https://cursor.com/dashboard?tab=bugbot) for commit c9090b89508e6859b58821a2fdad75e7a55a19db. This will update automatically on new commits. Configure [here](https://cursor.com/dashboard?tab=bugbot).</sup>

### CODE REVIEW

File: packages/codex-runner/src/CodexRunner.ts [hunk 1]
```diff
@@ -0,0 +1,518 @@
...
+		const prompt = (stringPrompt ?? streamingInitialPrompt ?? "").trim();
+		const threadOptions = this.buildThreadOptions();
+		const codex = this.createCodexClient();
+		const thread = this.config.resumeSessionId
+			? codex.resumeThread(this.config.resumeSessionId, threadOptions)
+			: codex.startThread(threadOptions);
+		const abortController = new AbortController();
+		this.abortController = abortController;
+
+		this.emitSystemInitMessage(sessionId);
```
[the remark this answers is not in the document]
Comment (Connoropolous) [reply packages/codex-runner/src/CodexRunner.ts]: Addressed in `f31bf51c`: `CodexRunner` now emits the `system:init` message only after `thread.started` (using the real `thread_id`), and falls back to emitting init at finalize only if startup fails before that event. This ensures persisted `codexSessionId` matches the actual thread for resume flows.

**Comment by Connoropolous** [comment]
### Reviewer Guide (1/6): Suggested Read Order

This PR is easier to review as six slices, in order:

1. Selector semantics (`[agent=...]`, `[model=...]`) and precedence:
https://github.com/ceedaragents/cyrus/pull/850/files#diff-d115d388854ec3ed02c47fb3b19e6f18536977e018441e836968b79e3aa3ccb3
2. Selector test matrix:
https://github.com/ceedaragents/cyrus/pull/850/files#diff-8bccb8a286bb77e2410336da4740b59eb4a7218912c4a5fc53af18a0854d43a9
3. Codex SDK runner + event projection:
https://github.com/ceedaragents/cyrus/pull/850/files#diff-5827c394746f6db98c8a2909b0575ded4e6bac173c9a7df8fe84367c61b9a254
4. Codex tool-event tests + Linear activity integration tests:
https://github.com/ceedaragents/cyrus/pull/850/files#diff-eb3e5fb97807440cd35d8fd23d26fe0817df3ffe49a598f0a4b57f0f79fc2a4d
https://github.com/ceedaragents/cyrus/pull/850/files#diff-e4d660023b08598c740f15ee08faf26f711af6dcde4bf8695dce9d5a4d62eae3
5. Formatter behavior (including todo checklist markdown):
https://github.com/ceedaragents/cyrus/pull/850/files#diff-dc24a3e4a55f0cc8c2754abed1c78faa10746ff60fde821ca2d1912bd8b5eec3
https://github.com/ceedaragents/cyrus/pull/850/files#diff-78ea3f4e367b91483f097f6c9de2e27a15038769ac414d70cdd776217f75568c
6. Config backward compatibility and default-model split:
https://github.com/ceedaragents/cyrus/pull/850/files#diff-f3f6049d68cdc11a5b2e2ae5773903d378ba68372e7f16c779d9c0128b23e745
https://github.com/ceedaragents/cyrus/pull/850/files#diff-a7f340765331d4f90cd433574daa73814a205c437684e63378937b53136c440a
https://github.com/ceedaragents/cyrus/pull/850/files#diff-c449fd2c759fdfd831bb2ec6c0c1b9248d05556418ef73bc19219790e58aeaa9
https://github.com/ceedaragents/cyrus/pull/850/files#diff-895f77ba4a2bfda15318a21bdc3ad8455e3b4309b3edb8b028301ce38e66d201

**Comment by Connoropolous** [comment]
### Reviewer Note (2/6): `[agent=...]` and `[model=...]` are cross-runner, not Codex-only

The selector behavior here applies to Claude, Gemini, and Codex:

- `[agent=claude|gemini|codex]` explicitly selects runner family.
- `[model=<name>]` selects model and can infer runner when agent is absent.
- Description tags override labels.
- If explicit agent conflicts with model family, explicit agent wins and model falls back to that runner’s default.

Implementation:
https://github.com/ceedaragents/cyrus/pull/850/files#diff-d115d388854ec3ed02c47fb3b19e6f18536977e018441e836968b79e3aa3ccb3

Coverage:
https://github.com/ceedaragents/cyrus/pull/850/files#diff-8bccb8a286bb77e2410336da4740b59eb4a7218912c4a5fc53af18a0854d43a9

**Comment by Connoropolous** [comment]
### Reviewer Note (3/6): Codex runner now projects SDK events into Cyrus tool lifecycle messages

This is the main reason Codex tool activity appears correctly in Linear now.

`CodexRunner` consumes streamed SDK events and projects non-chat items into `tool_use` + `tool_result` pairs (same contract expected by the existing session pipeline), including:

- `command_execution` -> `Grep`/`Glob`/`Read`/`Write`/`Bash`
- `file_change` -> `Edit`
- `web_search` (`open_page`) -> `WebFetch` (otherwise `WebSearch`)
- `mcp_tool_call` -> `mcp__<server>__<tool>`
- `todo_list` -> `TodoWrite`

It also de-duplicates `item.started`/`item.completed` by tool-use id.

Implementation:
https://github.com/ceedaragents/cyrus/pull/850/files#diff-5827c394746f6db98c8a2909b0575ded4e6bac173c9a7df8fe84367c61b9a254

Coverage:
https://github.com/ceedaragents/cyrus/pull/850/files#diff-eb3e5fb97807440cd35d8fd23d26fe0817df3ffe49a598f0a4b57f0f79fc2a4d

**Comment by Connoropolous** [comment]
### Reviewer Note (4/6): End-to-end visibility path for Codex tool actions in Linear

The behavior change is not just local formatting: projected Codex tool messages now flow through `AgentSessionManager` and become Linear action entries.

Integration test validates both action parameter and action result activity for a real projected `file_change` event:

https://github.com/ceedaragents/cyrus/pull/850/files#diff-e4d660023b08598c740f15ee08faf26f711af6dcde4bf8695dce9d5a4d62eae3

**Comment by Connoropolous** [comment]
### Reviewer Note (5/6): Formatter targets UI output contract and fixes checklist markdown

`IMessageFormatter` output is timeline/UI-facing text (not model input). This PR updates Codex formatting to match that target, including valid markdown task list output for todos (`- [ ]` / `- [x]`).

Formatter:
https://github.com/ceedaragents/cyrus/pull/850/files#diff-dc24a3e4a55f0cc8c2754abed1c78faa10746ff60fde821ca2d1912bd8b5eec3

Formatter tests:
https://github.com/ceedaragents/cyrus/pull/850/files#diff-78ea3f4e367b91483f097f6c9de2e27a15038769ac414d70cdd776217f75568c

**Comment by Connoropolous** [comment]
### Reviewer Note (6/6): Backward-compatible config migration for model defaults

Global defaults are split per runner (`claudeDefaultModel`, `geminiDefaultModel`, `codexDefaultModel`) while preserving compatibility for legacy `defaultModel` fields by migrating those to Claude-specific keys on load/update.

Schema:
https://github.com/ceedaragents/cyrus/pull/850/files#diff-f3f6049d68cdc11a5b2e2ae5773903d378ba68372e7f16c779d9c0128b23e745

Updater migration:
https://github.com/ceedaragents/cyrus/pull/850/files#diff-a7f340765331d4f90cd433574daa73814a205c437684e63378937b53136c440a

CLI migration path + tests:
https://github.com/ceedaragents/cyrus/pull/850/files#diff-c449fd2c759fdfd831bb2ec6c0c1b9248d05556418ef73bc19219790e58aeaa9
https://github.com/ceedaragents/cyrus/pull/850/files#diff-895f77ba4a2bfda15318a21bdc3ad8455e3b4309b3edb8b028301ce38e66d201
